### PR TITLE
Bump Zope packages to support Python 3.6

### PR DESCRIPTION
**Pull request by adisbladis**
pythonpackages.zope_testrunner: 4.4.10 -> 4.7.0
pythonpackages.zope_interface: 4.1.3 -> 4.4.2
pythonPackages.zconfig: 3.0.3 -> 3.2.0

###### Motivation for this change

This fixes build failures on Python 3.6

###### Things done

- [ ] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [x] Linux
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by FRidh** [comment]
Please move the expressions into separate files as explained in the head of `python-packages.nix`.

**Comment by lsix** [comment]
Hi,

Could each update also be done in a separate commit (one for `zope_testrunner`, one for `zope_interface` and finally one for `zconfig`)?

**Comment by adisbladis** [comment]
@FRidh I didn't want to do this for a simple version bump but sure, it has been done.
@lsix Done :)

File: pkgs/development/python-modules/zconfig/default.nix [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+, buildPythonPackage
+, pkgs
+, zope_testrunner
+}:
+
+buildPythonPackage rec {
+  name = "zconfig-${version}";
+  version = "3.2.0";
+
+  src = pkgs.fetchurl {
```
Comment (FRidh) [inline pkgs/development/python-modules/zconfig/default.nix]: `fetchPyPI`

File: pkgs/development/python-modules/zconfig/default.nix [hunk 2]
```diff
@@ -0,0 +1,27 @@
+{ stdenv
+, pythonPackages
+, buildPythonPackage
+, pkgs
+, zope_testrunner
+}:
+
+buildPythonPackage rec {
+  name = "zconfig-${version}";
```
Comment (FRidh) [inline pkgs/development/python-modules/zconfig/default.nix]:
```nix
pname = "ZConfig";
version = "3.2.0";
name = "${pname}-${version}";
```
We use the name as used on PyPI, so including uppercase.

File: pkgs/development/python-modules/zconfig/default.nix [hunk 3]
```diff
@@ -0,0 +1,27 @@
+{ stdenv
+, pythonPackages
```
Comment (FRidh) [inline pkgs/development/python-modules/zconfig/default.nix]: pass in individual items, not the whole package set

File: pkgs/development/python-modules/zconfig/default.nix [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+, buildPythonPackage
+, pkgs
```
Comment (FRidh) [inline pkgs/development/python-modules/zconfig/default.nix]: pass in individual items, not the whole package set

File: pkgs/development/python-modules/zconfig/default.nix [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+  version = "3.2.0";
+
+  src = pkgs.fetchurl {
+    url = "mirror://pypi/Z/ZConfig/ZConfig-${version}.tar.gz";
+    sha256 = "de0a802e5dfea3c0b3497ccdbe33a5023c4265f950f33e35dd4cf078d2a81b19";
+  };
+
+  propagatedBuildInputs = [ zope_testrunner ];
+
+  doCheck = false;
```
Comment (FRidh) [inline pkgs/development/python-modules/zconfig/default.nix]: Please include a comment explaining why you disable tests.

File: pkgs/development/python-modules/zope_testrunner/default.nix [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+buildPythonPackage rec {
+  name = "zope.testrunner-${version}";
+  version = "4.7.0";
+
+  src = fetchurl {
+    url = "mirror://pypi/z/zope.testrunner/${name}.zip";
+    sha256 = "8ffcb4989829544a83d27e42b2eeb28f8fc134bd847d71ce8dca54f710526ef0";
+  };
+
+  propagatedBuildInputs = [ zope_interface zope_exceptions zope_testing six ] ++ stdenv.lib.optional (!python.is_py3k or false) subunit;
```
Comment (FRidh) [inline pkgs/development/python-modules/zope_testrunner/default.nix]: `isPy3k` instead of `python.is_py3k`

**Review by FRidh — CHANGES_REQUESTED** [review]
The comments I made on the first package also apply to the other two.

**Comment by adisbladis** [comment]
@FRidh Ok, all packages updated.

File: pkgs/development/python-modules/zope_testrunner/default.nix [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+, six
+, subunit
+}:
+
+
+buildPythonPackage rec {
+  name = "zope.testrunner-${version}";
+  version = "4.7.0";
+
+  src = fetchurl {
```
Comment (FRidh) [inline pkgs/development/python-modules/zope_testrunner/default.nix]: fetchPypi

**Comment by FRidh** [comment]
cc maintainer @cillianderoiste

**Comment by FRidh** [comment]
Turns out that this caused quite a rebuild on Hydra, probably due to `zope.interface`.

**Pull request merged**